warehouse/oidc/models/activestate.py

```diff
@@ -0,0 +1,202 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import urllib
+
+from typing import Any
+
+from sqlalchemy import ForeignKey, String, UniqueConstraint
+from sqlalchemy.dialects.postgresql import UUID
+from sqlalchemy.orm import Query, mapped_column
+
+import warehouse.oidc.models._core as oidccore
+
+from warehouse.oidc.errors import InvalidPublisherError
+from warehouse.oidc.interfaces import SignedClaims
+from warehouse.oidc.models._core import (
+    CheckClaimCallable,
+    OIDCPublisher,
+    PendingOIDCPublisher,
+)
+
+ACTIVESTATE_URL = "https://platform.activestate.com"
+
+ACTIVESTATE_OIDC_API_PATH = "api/v1/oauth/oidc"
+
+
+def _check_sub(
+    ground_truth: str, signed_claim: str, _all_signed_claims: SignedClaims
+) -> bool:
+    # We expect a string formatted as follows:
+    #  org:<orgName>:project:<projectName>
+    # where :OPTIONAL-STUFF is a concatenation of other job context
+    # metadata. We currently lack the ground context to verify that
+    # additional metadata, so we limit our verification to just the ORG/REPO
+    # component.
```

Comment:
Nitpick: This comment looks stale/slightly off: there's no mention of `:OPTIONAL-STUFF` 🙂